packages/signing-utils/src/signing-clients/index.ts

```diff
@@ -0,0 +1,82 @@
+import type { ConnectConfig } from 'ssh2';
+
+import * as path from 'path';
+import { SSHClient } from '../ssh-client';
+import { LocalSigningClient } from './local-signing-client';
+import { RemoteSigningClient } from './remote-signing-client';
+
+export { LocalSigningClient } from './local-signing-client';
+export { RemoteSigningClient } from './remote-signing-client';
+
+export type SigningMethod = 'gpg' | 'jsign';
+
+export type SigningClientOptions = {
+  workingDirectory: string;
+  signingScript: string;
+  signingMethod: SigningMethod;
+};
+
+/** Options for signing a file remotely over an SSH connection. */
+export type RemoteSigningOptions = {
+  /** Username for authentication. */
+  username?: string;
+  /** Password for password-based user authentication. */
+  password?: string;
+  /** Port number of the ssh server. */
+  port?: number;
+  /** Buffer or string that contains a private key for either key-based or hostbased user authentication (OpenSSH format). */
+  privateKey?: Buffer | string;
+  /** The method to sign with.  Use gpg on linux and jsign on windows. */
+  signingMethod: SigningMethod;
+
+  /**
+   * The path of the working directory in which to sign files **on the remote ssh server**.  Defaults to `/home/ubuntu/garasign`.
+   */
+  workingDirectory?: string;
+  client: 'remote';
+};
+
+/** Options for signing a file locally. */
+export type LocalSigningOptions = {
+  /** The method to sign with.  Use gpg on linux and jsign on windows. */
+  signingMethod: SigningMethod;
+
+  client: 'local';
+};
+
+export type ClientOptions = RemoteSigningOptions | LocalSigningOptions;
+
+export interface SigningClient {
+  sign(file: string): Promise<void>;
+}
+
+export async function getSigningClient(
+  options: ClientOptions
+): Promise<SigningClient> {
+  async function getSshClient(sshOptions: ConnectConfig) {
+    const sshClient = new SSHClient(sshOptions);
+    await sshClient.connect();
+    return sshClient;
+  }
+
+  const signingScript = path.join(__dirname, '../..', 'src', './garasign.sh');
```

Comment:
Ignore if i'm wrong about this, but would this work on windows?

Reply:
yes, `/` works as a path separator on Windows just like `\` does :) so this could for example also be written as

```suggestion
  const signingScript = path.join(__dirname, '..', '..', 'src', 'garasign.sh');
```

or 

```suggestion
  const signingScript = path.join(__dirname, '../../src/garasign.sh');
```

and those would return the same result